### ISSUE TITLE
SVG image size should take the viewBox attribute into account

### ISSUE THREAD

**Issue by rubycon** [stylus/stylus#2183]
Currently SVG image size is only queried from the `width`/`height` attributes of the SVG files : 

- https://github.com/stylus/stylus/blob/dev/lib/functions/image.js#L149

However many SVG files define their dimensions with the `viewBox` attribute, or a combination of the `viewBox` and `width`/`height` attributes (in which case the two former attributes are relative to the first).
It would be nice if the SVG image size would also take viewBox into account. It would make the use of various SVG easier.

**Comment by alexeyten** [comment]
Just to be clear, width/height are not relative to viewBox.

On the other hand, width and height attributes could use length units such as `width="10pt"` or `height="10cm"` (I've never seen it in real life though) and this information is lost now.

**Comment by rubycon** [comment]
Yep sorry, the image is cropped and zoomed relative to the viewBox but the file size is still define by the `width` and `height` attributes if present. So if they aren't present Stylus should maybe check for viewBox values :

```javascript
if ('svg' == node.name)
  if(node.attributes.width && node.attributes.height) {
    width  = parseInt(node.attributes.width,  10);
    height = parseInt(node.attributes.height, 10);
  } else if(node.attributes.viewBox) {
    var viewBox = node.attributes.viewBox.split(' ');
    if(Object.keys(viewBox).length == 4) {
      width  = viewBox[2] - viewBox[0];
      height = viewBox[3] - viewBox[1];
    }
  }
```
The function could also check for more exotic units and convert them
https://www.w3.org/TR/SVG/coords.html#Units
NPM package already exist for this purpose : 
https://www.npmjs.com/package/svg-units
